src/main/java/org/jenkinsci/plugins/docker/workflow/DockerUtils.java

```diff
@@ -52,12 +60,19 @@ private DockerUtils() {
 
                 String parts[] = keyVal.split("=", 2);
                 if (parts.length != 2) {
-                    throw new IllegalArgumentException("Illegal syntax for --build-arg " + keyVal + ", need KEY=VALUE");
+                    // check if we have an environment variable with the very same name
+                    if (env.get(parts[0]) != null) {
+                        String key = parts[0];
+                        String value = "";
```

Comment:
It doesn't seem like you're actually setting the value here. Maybe this:
```suggestion
                        String value = end.get(parts[0]);
```

Reply:
To be honest, I dont really have a clue on when and where this map is actually used. :->

To me it seems it is NOT used for running the actual docker build subshell?

In reality the argument to docker build ist really just '--build-arg FOO', and docker itself will pickup $FOO from environment. docker itself won't do any complaints if $FOO is missing. 

So it seemed most reasonable for me in DockerUtils to add an empty string to the map und supress the Exception when there is a proper variable in the environment.

I definitly don't want to end up with running '--build-arg FOO=$FOO'. ;)

Does that clearify some things?
Would you suggest to add the value of the environment variable to the map?

Reply:
Right, I understand you. I had assumed that the intention was to replicate Docker CLI's functionality, I hadn't realised that jenkins shells out to Docker do this processing for it.

Would it be better to add a higher-level functional test for this behaviour? As it is the test you've got doesn't verify that the expected environment variable is injected correctly, only that there exists a key in a map.

Reply:
Thx for getting back on this. Not quite sure. 'docker build --build-arg FOO' is a perfectly valid docker build sequence, I assume we don't really want to test that this is working like it should as that is in the scope of docker. 

We just added some kind of sanity check wether such a variable is available in the environment and refuse to run the build if this is not the case. Actually I think docker build probably would be the better place to run this check.

So, in reality we relaxed the narrow KEY=VALUE requirement of the FromFingerprintStep to allow KEY as well under certain circumstances.